ubuntu/standard/Dockerfile.template

```diff
@@ -149,3 +154,21 @@ RUN apt-get update \
  && apt-get install -y --no-install-recommends \
     subversion \
  && rm -rf /var/lib/apt/lists/*
+
+# Download hosted tool cache
+ENV AGENT_TOOLSDIRECTORY=/opt/hostedtoolcache
+RUN azcopy --recursive --source https://vstsagenttools.blob.core.windows.net/tools/hostedtoolcache/linux --destination $AGENT_TOOLSDIRECTORY
+
+# Install the tools from the hosted tool cache
+RUN cd $AGENT_TOOLSDIRECTORY \
+ && for tool in $(ls); do \
```

Comment:
If there is ever any chance that a tool name (or a version in line 166 below) could have characters in it that require quoting at the shell level, this will fail to correctly identify the items. For instance, if there are tools "foo", "my tool" and "bar", this line effectively expands to "for tool in foo my tool bar; do" so it will try to iterate over tools "foo", "my", "tool" and "bar", which is incorrect.

Reply:
You might consider using find here to locate all the setup.sh scripts, e.g. `find /opt/hostedtoolcache -name setup.sh`. Avoids the nested for loop.

Reply:
Sure, I'll use `find`.  As for the tool names, we control the tool cache layout, so I think it's fine to require tool names not to contain spaces.  I don't know of an easy way to get around this in the script.